atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/InstrumentedCqlExecutor.java

```diff
@@ -0,0 +1,111 @@
+import com.palantir.processors.AutoDelegate;
+
+@AutoDelegate(typeToExtend = CqlExecutor.class)
+public class InstrumentedCqlExecutor extends AutoDelegate_CqlExecutor {
+    private static final String SERVICE_NAME = "cassandra-cql-executor";
+    private static final MetricRegistry METRIC_REGISTRY = AtlasDbMetrics.getMetricRegistry();
+
+    private static final Timer GET_TIMESTAMPS_TIMER = METRIC_REGISTRY.timer(
+            MetricRegistry.name(CqlExecutor.class, "getTimestamps"));
+    private static final Timer GET_TIMESTAMPS_WITHIN_ROW_TIMER = METRIC_REGISTRY.timer(
+            MetricRegistry.name(CqlExecutor.class, "getTimestampsWithinRow"));
+
+    private CqlExecutor delegate;
+
+    public InstrumentedCqlExecutor(CqlExecutor delegate) {
+        super(null, null);
+        this.delegate = delegate;
+    }
+
+    @Override
+    public CqlExecutor delegate() {
+        return delegate;
+    }
+
+    @Override
+    public List<CellWithTimestamp> getTimestamps(TableReference tableRef, byte[] startRowInclusive, int limit) {
+        // TODO(ssouza): also log the row key names when they can be marked as safe for logging.
+
+        // noinspection unused - try-with-resources closes trace
+        try (CloseableTrace trace = startLocalTrace("cqlExecutor.getTimestamps(table {}, limit {})",
+                LoggingArgs.safeTableOrPlaceholder(tableRef), limit)) {
+            return registerDuration(() ->
+                    delegate.getTimestamps(tableRef, startRowInclusive, limit), GET_TIMESTAMPS_TIMER,
+                    logger -> logger.warn("cqlExecutor.getTimestamps({}, {})",
+                            LoggingArgs.tableRef(tableRef),
+                            SafeArg.of("limit", limit)));
+        }
+    }
+
+    @Override
+    public List<CellWithTimestamp> getTimestampsWithinRow(TableReference tableRef, byte[] row,
+            byte[] startColumnInclusive, long startTimestampExclusive, int limit) {
+        // TODO(ssouza): also log the row key names when they can be marked as safe for logging.
+
+        //noinspection unused - try-with-resources closes trace
+        try (CloseableTrace trace = startLocalTrace("cqlExecutor.getTimestampsWithinRow(table {}, ts {}, limit {})",
+                LoggingArgs.safeTableOrPlaceholder(tableRef), startTimestampExclusive, limit)) {
+            return registerDuration(() -> delegate.getTimestampsWithinRow(
+                    tableRef, row, startColumnInclusive, startTimestampExclusive, limit),
+                    GET_TIMESTAMPS_WITHIN_ROW_TIMER,
+                    logger -> logger.warn("cqlExecutor.getTimestampsWithinRow({}, {}, {})",
+                            LoggingArgs.tableRef(tableRef),
+                            SafeArg.of("sweepTs", startTimestampExclusive),
+                            SafeArg.of("limit", limit)));
+        }
+    }
+
+    private <T> T registerDuration(Supplier<T> callable, Timer timer, Consumer<Logger> loggerFunction) {
```

Comment:
This is _almost_ identical to `InstrumentedCassandraClient.registerDuration`. Can we (post merge if it's not a 2-minute task) deduplicate the code?